packages/graphiql/src/YogaGraphiQL.tsx

```diff
@@ -181,7 +181,12 @@ export function YogaGraphiQL(props: YogaGraphiQLProps): React.ReactElement {
           defaultVariableEditorOpen={true}
           docExplorerOpen={showDocs}
           onToggleDocs={() => setShowDocs((isOpen) => !isOpen)}
-          tabs
+          tabs={{
+            onTabChange: (tab) => {
+              // recycle the request headers within all tabs
+              tab.tabs.forEach((t) => (t.headers = props.headers))
+            },
+          }}
```

Comment:
I suppose this runs every render? not sure if there is a better way to do this 🤔

Reply:
I think it makes sense to run on every render because if the headers in the props change, we want all the tabs to change right?

Reply:
Tangential question: On line 115, is there a reason we override the prop's header with an empty array, then add the props.header to extensions on line 128? This results in my custom headers being set in body of the request rather than the headers of the event. 

There's a bigger picture your team has that I don't so curious on your thoughts!

```
{
  httpMethod: 'POST',
 headers: {
     ..... // is it possible to get the custom headers here?
  },
   ....
  body: '{"query":"query Redwood {\\n  redwood {\\n version\\n  }","variables":null, "operationName":"Redwood", 
 "extensions":{"headers":{"cookie":"G_AUTHUSER_H=0; session=...","auth-provider":"dbAuth","authorization":"Bearer 1"}}}',
}
```

Reply:
> On line 115, is there a reason we override the prop's header with an empty array, then add the props.header to extensions on line 128

I am unsure I think we should be able to do and introspection from URL loader gets the headers. @ardatan thoughts?
```
headers: JSON.parse(props.headers ?? '{}'),
  ```

Reply:
Yes, that is the solution I was thinking of as well... as long as there's no reason we need the empty header. 

If you can't make this change in this PR, I'm happy to open up a new one.

Reply:
https://github.com/dotansimha/graphql-yoga/pull/1214